wgsl/index.bs

```diff
@@ -3412,6 +3412,39 @@ TODO: *Stub*. Call to function that has a [=return type=] is an expression.
 
 TODO: *Stub*: how to write each of the abstract pointer operations
 
+## Constant Identifier Expression  ## {#constant-identifier-expr}
+
+<table class='data'>
+  <caption>Getting the value of a `let`-declared identifier</caption>
+  <thead>
+    <tr><th>Precondition<th>Conclusion<th>Description
+  </thead>
+  <tr algorithm="pipeline-overridable constant value">
+       <td>
+          |c| is an identifier [=resolves|resolving=] to
+          an [=in scope|in-scope=] [=pipeline-overridable=] `let` declaration with type |T|
+       <td class="nowrap">
+          |c| : |T|
+       <td>If pipeline creation specified a value for the constant ID, then the result is
+           that value.<br>
+           Otherwise, the result is the value computed for the initializer expression.<br>
+           Note: pipeline-overridable constants appear at module-scope, so evaluation occurs
+           before the shader begins execution.
+  <tr algorithm="constant value">
+       <td>
+          |c| is an identifier [=resolves|resolving=] to
+          an [=in scope|in-scope=] `let` declaration with type |T|,
+          and is not pipeline-overridable
+       <td class="nowrap">
+          |c| : |T|
+       <td>Result is the value computed for the initializer expression when the `let` declaration
+           was executed.<br>
+           For a `let` declaration at module scope, that occurs before the shader begins execution.<br>
```

Comment:
Interesting. Do we have to say it's actually "executed"? I'd expect, intuitively, that there is no execution here. It's just a value that is named. I.e. it may as well be executed at shader creation time.

Reply:
That's fair, I've reworded like this:

       <td>Result is the value computed for the initializer expression.<br>
           For a `let` declaration at module scope, evaluation occurs before the shader begins execution.<br>
           For a `let` declaration inside a function, evaluation occurs each time control reaches
           the declaration.<br>